Null safety Support

nischalkumar opened the pull request:
- Migrated dependencies to null safety dependencies
- migrated Flat Button deprecated widget To TextButton
- Migrate a few params to optional

milhomem commented:
Hi @nischalkumar Thank you very much for your contributions, they are awesome.

Would you mind to separate 
- Migrated dependencies to null safety dependencies
- migrated Flat Button deprecated widget To TextButton
- Migrate a few params to optional

into 3 separate commits?

Also, on your commit messages if you can put the WHY you are doing the changes it will help a lot to review and to make decisions in the future.

Like, although I like the changes I can't say why you:
- migrated Flat Button deprecated widget To TextButton

or why you:
- Migrate a few params to optional

or why this changed:
- if (_updateInfo?.updateAvailable == true) {
+      if (_updateInfo.updateAvailability == UpdateAvailability.updateAvailable) {

nischalkumar commented:
@milhomem Thank you for the comments.
Since to change commit history for this pr, I would be required to do a force push, instead, I have raised another pr: [](https://github.com/ofload/native_updater/pull/20). 
which addresses all the comments mentioned in this pr. I have also added comments addressing each concern.

Marking this pr as closed and please look at pr 20.
